generatedocs/pkg/doccompiler/checks.go

```diff
@@ -61,22 +58,23 @@ func checkAllConfigFieldsDocumented(data *docData) []error {
 	for _, field := range expectedConfigFields {
 		docEntry, ok := data.ConfigMap[field]
 		if !ok {
-			findings = append(findings, errors.Errorf(`config field "%s" is not documented`, field))
+			findings = errors.Join(findings, fmt.Errorf(`config field "%s" is not documented`, field))
 			continue
 		}
 
 		if docEntry.Description == "" {
-			findings = append(findings, errors.Errorf(`config field "%s" has no description`, field))
+			findings = errors.Join(findings, fmt.Errorf(`config field "%s" has no description`, field))
 		}
 		if docEntry.Examples == "" {
-			findings = append(findings, errors.Errorf(`config field "%s" has no examples`, field))
+			findings = errors.Join(findings, fmt.Errorf(`config field "%s" has no examples`, field))
 		}
 	}
+
 	return findings
```

Comment:
A single `errors.Join` at the end would create a flat structure that is easier to work with calling `Unwrap` on it.

Reply:
ok, not sure what you mean, how do I add them to the findings without joining?

Reply:
Unwrap always return `nil` on joined errors btw so not good idea to use. Commited a version with slightly reduced join nesting at least. Result print still same independent of join nesting.